billing-tests/src/androidTest/java/com/mapbox/navigation/billing/test/BillingTests.kt

```diff
@@ -0,0 +1,156 @@
+import com.mapbox.navigation.testing.ui.utils.coroutines.sdkTest
+import com.mapbox.navigation.testing.ui.utils.coroutines.setNavigationRoutesAndWaitForUpdate
+import com.mapbox.navigation.testing.ui.utils.coroutines.startTripSessionAndWaitForActiveGuidanceState
+import com.mapbox.navigation.testing.ui.utils.getMapboxAccessTokenFromResources
+import com.mapbox.navigation.testing.ui.utils.runOnMainSync
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.take
+import kotlinx.coroutines.flow.toList
+import org.junit.After
+import org.junit.Assert
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import java.net.URI
+import java.util.concurrent.CancellationException
+import java.util.concurrent.TimeUnit
+
+class BillingTests : BaseCoreNoCleanUpTest() {
+
+    @get:Rule
+    val mapboxNavigationRule = MapboxNavigationRule()
+
+    private lateinit var mapboxNavigation: MapboxNavigation
+
+    private lateinit var httpEventsObserver: HttpServiceEventsObserver
+
+    private val source = Point.fromLngLat(-77.031991, 38.894721)
+    private val destination = Point.fromLngLat(-77.030923, 38.895433)
+
+    override fun setupMockLocation(): Location = mockLocationUpdatesRule.generateLocationUpdate {
+        latitude = source.latitude()
+        longitude = source.longitude()
+        bearing = 190f
+    }
+
+    @Before
+    fun setup() {
+        httpEventsObserver = HttpServiceEventsObserver.register()
+
+        runOnMainSync {
+            assertTrue(httpEventsObserver.billingRequests().isEmpty())
+
+            mapboxNavigation = MapboxNavigationProvider.create(
+                NavigationOptions.Builder(context)
+                    .accessToken(getMapboxAccessTokenFromResources(context))
+                    .routingTilesOptions(
+                        RoutingTilesOptions.Builder()
+                            .tilesBaseUri(URI(mockWebServerRule.baseUrl))
+                            .build()
+                    )
+                    .build()
+            )
+        }
+    }
+
+    @After
+    fun cleanUp() {
+        HttpServiceEventsObserver.unregister()
+    }
+
+    @Test
+    fun test30SecondsLongActiveGuidanceTrip() = sdkTest(TimeUnit.MINUTES.toMillis(1)) {
+        val routes = requestRoutes(source, destination)
+
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(routes)
+        mapboxNavigation.startTripSessionAndWaitForActiveGuidanceState()
+
+        delay(TimeUnit.SECONDS.toMillis(30))
+
+        mapboxNavigation.clearNavigationRoutesAndWaitForUpdate()
+
+        val billingRequests = getAllBillingEventsWhenHasEvents(1, "Active Guidance") {
+            it.isActiveGuidanceSession
+        }
+
+        assertEquals(1, billingRequests.size)
+        assertTrue(billingRequests.first().isActiveGuidanceSession)
+        assertSessionDuration(30, billingRequests.first().duration)
```

Comment:
Why is it e2e test? It's the same as previous tests, except for the duration. Or isn't it?

Reply:
I've added comments for this test.